### PR TITLE
Master

### PR DESCRIPTION
**Pull request by TOMASHEVSKIYIVAN**
## Summary
fixed usbconsole on weact STM32H743VI
## Impact
new debug experience for users

### CODE REVIEW

**Review by buzdalin — COMMENTED** [review]
usbconsole weact STM32H743VI fix

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)

__Fill In The Commit Message:__ This PR contains a Commit with an Empty Commit Message. Please fill in the Commit Message with the PR Summary.



**No, this PR does not fully meet the NuttX requirements.** Here's why:

* **Insufficient Summary:** The summary is too brief.  
    * **What was the problem with usbconsole?**  
    * **How was it fixed? What code was changed?**
* **Impact:** While a "new debug experience" is mentioned, it's not clear how it impacts:
    * **Users:** Will they need to do anything differently?
    * **Build:** Are there any configuration changes needed?
    * **Hardware:** Are there specific USB requirements now?
    * **Documentation:** Is documentation updated to reflect the fix?
* **Missing Testing Details:**  
    * **No build host information provided.**
    * **No "before" and "after" testing logs.**

**To improve this PR:**

1. **Expand the Summary:** Provide specific details about the bug and the fix.
2. **Complete the Impact Section:** Address all the points mentioned above. 
3. **Add Testing Details:** Specify your build host environment and include the requested logs.

File: arch/arm/src/stm32h7/Kconfig [hunk 1]
```diff
@@ -616,7 +616,7 @@ config STM32H7_HAVE_GPIOF
 
 config STM32H7_HAVE_GPIOG
 	bool
-	default n
+	default y
```
Comment (acassis) [inline arch/arm/src/stm32h7/Kconfig:619]: none STM32H7_HAVE_GPIOx should be enabled by default, this symbol is enabled conditionally by the MCU family

File: arch/arm/src/stm32h7/hardware/stm32h7x3xx_pinmap.h [hunk 2]
```diff
@@ -728,10 +728,10 @@
...
 /* USB on-the-go full-speed (OTG_FS) */
 
-#define GPIO_OTGFS_DM_0           (GPIO_ALT|GPIO_AF10|GPIO_PUSHPULL|GPIO_FLOAT|GPIO_PORTA|GPIO_PIN11)
-#define GPIO_OTGFS_DP_0           (GPIO_ALT|GPIO_AF10|GPIO_PUSHPULL|GPIO_FLOAT|GPIO_PORTA|GPIO_PIN12)
-#define GPIO_OTGFS_ID_0           (GPIO_ALT|GPIO_AF10|GPIO_OPENDRAIN|GPIO_PULLUP|GPIO_PORTA|GPIO_PIN10)
-#define GPIO_OTGFS_SOF_0          (GPIO_ALT|GPIO_AF10|GPIO_PUSHPULL|GPIO_FLOAT|GPIO_PORTA|GPIO_PIN8)
+#define GPIO_OTGFS_DM           (GPIO_ALT|GPIO_AF10|GPIO_PUSHPULL|GPIO_FLOAT|GPIO_PORTA|GPIO_PIN11)
+#define GPIO_OTGFS_DP           (GPIO_ALT|GPIO_AF10|GPIO_PUSHPULL|GPIO_FLOAT|GPIO_PORTA|GPIO_PIN12)
+#define GPIO_OTGFS_ID           (GPIO_ALT|GPIO_AF10|GPIO_OPENDRAIN|GPIO_PULLUP|GPIO_PORTA|GPIO_PIN10)
+#define GPIO_OTGFS_SOF          (GPIO_ALT|GPIO_AF10|GPIO_PUSHPULL|GPIO_FLOAT|GPIO_PORTA|GPIO_PIN8)
```
Comment (acassis) [inline arch/arm/src/stm32h7/hardware/stm32h7x3xx_pinmap.h:734]: That is also incorrect, the new pinmap format requires the _0 at ending and you need to define the GPIO_OTGFS_DM at your board.h like this:
```
#define GPIO_OTGFS_DM  (GPIO_OTGFS_DM_0|GPIO_SPEED_100MHz) /* PA11 */
```

Please use the linum-stm32h753bi as reference

File: boards/arm/stm32h7/weact-stm32h743/src/weact-stm32h743.h [hunk 3]
```diff
@@ -36,6 +36,13 @@
 
 /* Configuration ************************************************************/
 
+#define HAVE_PROC            0
+#define HAVE_USBDEV          1
+#define HAVE_USBHOST         1
+#define HAVE_USBMONITOR      0
+#define HAVE_MTDCONFIG       0
+#define HAVE_PROGMEM_CHARDEV 0
```
Comment (acassis) [inline boards/arm/stm32h7/weact-stm32h743/src/weact-stm32h743.h:44]: Normally all HAVE_xxx should be defined as 1 and undefined case some criteria (i.e.: CONFIG_ symbol missing) is not met

**Comment by acassis** [comment]
Hi @buzdalin thank you for your contribution. Please fix my comments.

You can read the documentation explaining how to fix and push the modifications to this PR:
https://nuttx.apache.org/docs/latest/contributing/making-changes.html#how-to-include-the-suggestions-on-your-pull-request

**Comment by acassis** [comment]
Hi @TOMASHEVSKIYIVAN, thanks for fixing the comments.

Please don't use Merge, use git rebase, as explained in the documentation. Also squash the PRs as explained in the link I sent to you early

**Comment by acassis** [comment]
The CI is failing because this small issue:

```
../nuttx/tools/checkpatch.sh -u -m -g 203944b7240d9c0a5deeb71a925688449024ced2..HEAD
Error: .../nuttx/boards/arm/stm32h7/weact-stm32h743/include/board.h:346:0: error: Wrong column position or missing blank line before comment
Error: Process completed with exit code 1.
```

**Comment by TOMASHEVSKIYIVAN** [comment]
@acassis, please excuse me. I'm kinda newbie in git and github

**Comment by TOMASHEVSKIYIVAN** [comment]
@acassis guess i created new pr...

**Comment by acassis** [comment]
> @acassis guess i created new pr...

Seems better, but you forgot to fix:

Error: /home/runner/work/nuttx/nuttx/nuttx/boards/arm/stm32h7/weact-stm32h743/include/board.h:346:0: error: Wrong column position or missing blank line before comment

You can test it before push the commit, this way:

./tools/checkpatch.sh -g HEAD~...HEAD

**Comment by acassis** [comment]
> @acassis, please excuse me. I'm kinda newbie in git and github

Don't worry! You are learning by doing!

**Comment by acassis** [comment]
@TOMASHEVSKIYIVAN now it is passing in the CI, but you forgot to squash the PR :-)

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**